src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -778,7 +778,11 @@ GenTree* Compiler::getArgForHWIntrinsic(var_types            argType,
             {
                 arg = impSIMDPopStack();
             }
+#if defined(TARGET_ARM64) && defined(FEATURE_MASKED_SIMD)
+            assert(varTypeIsSIMD(arg) || varTypeIsMask(arg));
+#else
             assert(varTypeIsSIMD(arg));
+#endif // TARGET_ARM64 && FEATURE_MASKED_SIMD
```

Comment:
This, and several other changes, likely should not be ARM64 specific.

`xarch` and `arm64` both support `TYP_MASK` and both generally need the same support around locals and other handling. Other platforms may have similar masking support in the future as well.

It would be ideal if such changes were added more generally, assuming it doesn't cause significant work outside the ARM64 areas of the codebase.

Reply:
> It would be ideal if such changes were added more generally

I've moved some of these to be generic. I think everything that's left is Arm specific.